ingestion/src/metadata/ingestion/source/snowflake_usage.py

```diff
@@ -60,8 +57,10 @@ class SnowflakeUsageSource(UsageSource):
 
     def __init__(self, config: WorkflowSource, metadata_config: OpenMetadataConnection):
         super().__init__(config, metadata_config)
-        start, end = get_start_and_end(self.config.sourceConfig.config.queryLogDuration)
-        end = end + timedelta(days=1)
+        # Snowflake allows retrieving data as far as 7 days, counting today
+        start, end = get_start_and_end(
+            min(self.config.sourceConfig.config.queryLogDuration, 6)
```

Comment:
Do you see any potential problems that we are returning today based on UTC time in `get_start_and_end`? 

```
    today = datetime.utcnow()
```

Reply:
As different users might be working with different timezones, we default to UTC for all date-management stuff. I think that just in case I added +1 to today to make sure we pick up the latest queries